source/session.ts

```diff
@@ -624,7 +628,7 @@ export class Session {
         throw this.getErrorFromResponse(response);
       }
       try {
-        return this.decode(response, {}, decodeDatesAsIso);
+        return this.decode(response, {}, decodeDatesAsIso || this.decodeDatesAsIso);
```

Comment:
This would force `decodeDatesAsIso` to true if it were set to true globally, we want the global value to be overridden by the local value if there is one.

Reply:
Oh, right... What was I thinking 🤣 I'll fix it Monday (since I am on a factory tour tomorrow), but feel free to swap them and merge before then.